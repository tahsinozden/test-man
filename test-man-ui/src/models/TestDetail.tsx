class TestDetail {
    constructor(public id?: number, public name?: string, public status?: string, public description?: string) {
    }
}

export default TestDetail;