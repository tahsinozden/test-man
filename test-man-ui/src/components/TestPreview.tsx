import React from "react";
import {Col, Form, Row} from "react-bootstrap";
import TestDetail from "../models/TestDetail";
import StatusDropdown from "./StatusDropdown";
import TestStatus from "../models/TestStatus";

interface TestPreviewProps {
    testDetail: TestDetail;
    onStatusSelect: (testDetail: TestDetail, newStatus: string) => void;
}

const TestPreview: React.FC<TestPreviewProps> = ({testDetail, onStatusSelect}: TestPreviewProps) => {
    const handleStatusSelect = (testDetail: TestDetail, newStatus: string) => {
        onStatusSelect(testDetail, newStatus);
    }

    return (
        <>
            <Form>
                <Row>
                    <Col>{testDetail.name}</Col>
                    <Col>
                        <StatusDropdown currentStatus={testDetail.status || TestStatus.Undefined.toString()}
                                        onStatusSelect={(newStatus) => handleStatusSelect(testDetail, newStatus)}/>
                    </Col>
                    <Col></Col>
                    <Col></Col>
                    <Col></Col>
                </Row>
            </Form>
        </>


    );
};

export default TestPreview;