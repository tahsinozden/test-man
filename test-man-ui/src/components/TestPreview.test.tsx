import renderer from "react-test-renderer";
import TestPreview from "./TestPreview";
import TestDetail from "../models/TestDetail";
import "../jestGlobalMocks"

it('renders correctly when there is a test detail', () => {
    const testDetail = new TestDetail(1, "name", "Passed", "");
    const tree = renderer
        .create(<TestPreview onStatusSelect={() => {
        }} testDetail={testDetail}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a test detail with undefined props', () => {
    const testDetail = new TestDetail();
    const tree = renderer
        .create(<TestPreview onStatusSelect={() => {
        }} testDetail={testDetail}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when there is no status present', () => {
    const testDetail = new TestDetail(1, "name", undefined, "");
    const tree = renderer
        .create(<TestPreview onStatusSelect={() => {
        }} testDetail={testDetail}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});