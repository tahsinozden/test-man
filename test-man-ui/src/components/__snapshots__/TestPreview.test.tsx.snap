// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a test detail 1`] = `
<form
  className=""
/>
`;

exports[`renders correctly when there is a test detail with undefined props 1`] = `
<form
  className=""
>
  <div
    className="row"
  >
    <div
      className="col"
    />
    <div
      className="col"
    >
      <div
        className="dropdown"
      >
        <button
          aria-expanded={false}
          className="dropdown-toggle btn btn-secondary"
          disabled={false}
          id="dropdown-item-button"
          onClick={[Function]}
          type="button"
        >
          Undefined
        </button>
      </div>
    </div>
    <div
      className="col"
    />
    <div
      className="col"
    />
    <div
      className="col"
    />
  </div>
</form>
`;

exports[`renders correctly when there is no status present 1`] = `
<form
  className=""
/>
`;
